keps/sig-node/compute-device-assignment.md

```diff
@@ -0,0 +1,153 @@
+
+[Device Monitoring](https://docs.google.com/document/d/1NYnqw-HDQ6Y3L_mk85Q3wkxDtGNWTxpsedsgw4NgWpg/edit?usp=sharing) requires external agents to be able to determine the set of devices in-use by containers and attach pod and container metadata for these devices.
+
+## Objectives
+
+* To enable node monitoring agents to determine the set of devices that are in-use, and to obtain metadata about the container/pod using them.
+* To enable consumption of network devices without direct coordination between CNI and Device Plugins.
+* To enable pre-assignment of devices at scheduling time.
+
+## Non-Objectives
+
+*  To define end-to-end model and user-story for network device plugins. Networks are pod-level objects but device assignment is container-level concept, therefore this requires further discussion outside the scope of this proposal. Though changes proposed here should be seen as a building block for designing and unblocking further innovations.
+*  To use pre-assignment of devices at scheduling time in core kubernetes. This is meant to enable experimentation for advanced use-cases out of tree.
+
+## User Journeys
+
+### Device Monitoring Agents
+
+* As a _Cluster Administrator_, I provide a set of devices from various vendors in my cluster.  Each vendor independently maintains their own agent, so I run monitoring agents only for devices I provide.  Each agent adheres to to the [node monitoring guidelines](https://docs.google.com/document/d/1_CdNWIjPBqVDMvu82aJICQsSCbh2BR-y9a8uXjQm4TI/edit?usp=sharing), so I can use a compatible monitoring pipeline to collect and analyze metrics from a variety of agents, even though they are maintained by different vendors.
+* As a _Device Vendor_, I manufacture devices and I have deep domain expertise in how to run and monitor them. Because I maintain my own Device Plugin implementation, as well as Device Monitoring Agent, I can provide consumers of my devices an easy way to consume and monitor my devices without requiring open-source contributions. The Device Monitoring Agent doesn't have any dependencies on the Device Plugin, so I can decouple monitoring from device lifecycle management. My Device Monitoring Agent works by periodically querying the `/pods` endpoint to discover which devices are being used, and to get the container/pod metadata associated with the metrics:
+
+![image](https://user-images.githubusercontent.com/11936386/42943622-e8c676f2-8b80-11e8-8d8f-96a016c6f3b4.png)
```

Comment:
That is not the case.  Since we expect devices to be identifiable by their device ID, this model will work for any device.  I will update the image, and see if we can construct a more detailed example of how monitoring would work for other devices.
Ill let @RenaudWasTaken to speak on the viability of this feature.
Ill add an alternatives section and cc you in a comment.